docs/how-to-work-on-the-component-library.md

```diff
@@ -0,0 +1,87 @@
+- Write unit tests
+
+## Researching and planning
+
+Before building a component, you need to research and document on how the existing version behaves and looks, to ensure that the new one has matching styles and supports all the current usages. In order to meet the web accesibility requirements, you should also pay attention to the accessibility aspect of the components, see which HTML elements and ARIA attributes are used under the hood.
+
+Once you have gathered enough information about the component, you can start thinking about the props interface. Ideally, the interface should be as similar to the current version as possible, to ease the adoption later on. Since we are using Bootstrap components, the simplest approach is to mimic [their implementation](https://github.com/react-bootstrap/react-bootstrap/tree/master/src).
+
+We prefer smaller pull requests rather than a large one, because they speed up the review time and reduce cognitive overload for the reviewers. For that reason, you should think about how you would break down the implementation and come up with a delivery plan.
+
+We recommend opening a separate GitHub issue for each component and include all the notes in the issue description. It can be used as a place to host all of your working notes, as well as a way to communicate the approach with the reviewers. We will use the issue thread for further discussion if needed. [The issue for Button component](https://github.com/freeCodeCamp/freeCodeCamp/issues/45357) can be used as a reference.
+
+## Implementing the component
+
+A new component can be created using the following command from the root directory:
+
+```bash
+cd tools/ui-components
+
+npm run gen-component MyComponent
+```
+
+The command will generate a new folder inside the `ui-components` directory, with the following files:
+
+| File name                  | Purpose                                                |
+| -------------------------- | ------------------------------------------------------ |
+| `index.ts`                 | It is used for exporting the component and its types.  |
+| `my-component.stories.tsx` | It is used for demoing the component on Storybook.     |
+| `my-component.test.tsx`    | It is a test file.                                     |
+| `my-component.tsx`         | It is where we implement the component.                |
+| `types.ts`                 | Is is where we locate component's interface and types. |
+
+Each component is different, but in general the components should:
+
+- Support forwarding ref
+- Be styled for both light and dark themes
+- Be styled internally based on their props (the consumers should not need to restyle the components with the `className` prop)
+- Utilize the built-in styling system from Tailwind instead of having custom styles
+
+### Useful links
+
+- [Tailwind CSS Configuration](https://tailwindcss.com/docs/configuration)
+- [React Bootstrap v0.33 Docs](https://react-bootstrap-v3.netlify.app/components/)
```

Comment:
this url : https://react-bootstrap-v3.netlify.app/components/ doesn't exist, maybe just https://react-bootstrap-v3.netlify.app/ is enough